study.md

```diff
@@ -21,18 +21,36 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
-```
+1. i dont't understand how callback functions are called back at a later time.
+2.  I don't understand an exampe of a function in an article:
+function example(a,b) {
+	number += a;
+	alert('You have chosen: ' + b);
+};
+what is the point of line 28 since the function just gives back an alert?
+
+3. I am confused about function declaration and expression
+is this a function decleration?:
+function foo() {
+ console.log('hello');
+}; and can be called above where it's defined because it's hoisted?
+
+and is this a function expression?:
+const x = function foo() {
+ console.log('hello');
+}; and it can only be called below it is defined? ```
```

Comment:
I wrote this example and hopefully it will help.
```javascript
// This will output, "[Function: sayHello]".
console.log(sayHello);

// This will work and output, "Hello".
sayHello();

// This will output, "undefined" since the declaration of "sayHi" has been
// hoisted to the top, but its definition has not been.
console.log(sayHi);

// This will throw an error for the reason above.  Since sayHi is undefined,
// it's trying to execute "undefined" as if it were a function.
sayHi();

// This gets hoisted to the top.
function sayHello() {
  console.log('Hello.');
}

// "var sayHi" gets hoisted to the top, but its assignment to function() { ... }
// does not.
var sayHi = function() {
  console.log('Hi.');
};

// This will output, "[Function: sayHi]", since it's after the definition of the
// function.
console.log(sayHi);

// This will output, "Hi".
sayHi();
```